Sources/NIOCore/ByteBuffer-core.swift

```diff
@@ -35,13 +35,15 @@ extension _ByteBufferSlice: Equatable {}
 /// fits within 24 bits, otherwise the behaviour is undefined.
 @usableFromInline
 struct _ByteBufferSlice {
-    @usableFromInline var upperBound: ByteBuffer._Index
-    @usableFromInline var _begin: _UInt24
+    @usableFromInline private(set) var upperBound: ByteBuffer._Index
+    @usableFromInline private(set) var _begin: _UInt24
```

Comment:
👍 Love that this is possible and we should use it more often.

Reply:
This is also a surprising performance win in some cases (though not here, I think). In particular it means that the compiler can exhaustively audit all code that is able to `set` this stored property, and so can often elide runtime exclusivity checks.